File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        #hash numbers to index
        #iterate through has table 
        #if target - current num in haah return list of indexes
        num_index = {}
        solution = []
        for  i in range(len(nums)):
                if (target - nums[i]) in num_index:
                    solution = [num_index[target-nums[i]], i]
                    return solution
                num_index[nums[i]] = i
    
        

       
        

File: Data Structures & Algorithms/duplicate-integer/submission-1.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        Set<Integer> numbers = new HashSet<>();
        for(int n: nums) {
            if(numbers.contains(n)){
                return true;
            }
            numbers.add(n);
            
        }
        return false;
    }
}
